### DISCUSSION TITLE
The previous git revision directories still remain

### DISCUSSION THREAD

**Discussion by jazzsir** [apache/airflow#20255] [General]
### Apache Airflow version

2.1.4

### What happened

After synchronizing a git repo, sometimes the previous git revision directory doesn't be removed with a file whose extension is "pyc"

### How to reproduce

This happens from time to time.

### Operating System

CentOS 7.6

### Deployment

Official Apache Airflow Helm Chart

### Deployment details

K8s: v1.16.7

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by potiuk** [comment]
You have not described what it is about, so I presume it is a git-sync that causes it. And there are no logs, examples etc. 
 We are using standard https://github.com/kubernetes/git-sync to perform the syncing. I am converting this into discussion as this is really not an Airlfow issue. I suggest that you provide some more details (such as examples where you saw it happening, details of your git-sync confiiguraio, and scenarions where you could show examples of the .pyc files remaining - and then I suggest you open an issue in git-sync repo with all those details.

**Reply by potiuk** [reply]
And in order for git-sync people to take a look, you shoudl really try to work out a reproducible scenario when it happens - otherwise they might have hard time to understand how to reproduce it.
